core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlFunctionBuilder.scala

```diff
@@ -0,0 +1,233 @@
+    "IDENT_SEED" -> FunDef(1, 1),
+    "SQL_VARIANT_PROPERTY" -> FunDef(2, 2),
+    "CURRENT_DATE" -> FunDef(0, 0),
+    "CURRENT_TIMESTAMP" -> FunDef(0, 0),
+    "CURRENT_TIMEZONE" -> FunDef(0, 0),
+    "CURRENT_TIMEZONE_ID" -> FunDef(0, 0),
+    "DATE_BUCKET" -> FunDef(3, 4),
+    "DATEADD" -> FunDef(3, 3),
+    "DATEDIFF" -> FunDef(3, 3),
+    "DATE_DIFF_BIG" -> FunDef(3, 3),
+    "DATEFROMPARTS" -> FunDef(3, 3),
+    "DATENAME" -> FunDef(2, 2),
+    "DATEPART" -> FunDef(2, 2),
+    "DATETIME2FROMPARTS" -> FunDef(8, 8),
+    "DATETIMEFROMPARTS" -> FunDef(7, 7),
+    "DATETIMEOFFSETFROMPARTS" -> FunDef(10, 10),
+    "DATETRUNC" -> FunDef(2, 2),
+    "DAY" -> FunDef(1, 1),
+    "EOMONTH" -> FunDef(1, 2),
+    "GETDATE" -> FunDef(0, 0),
+    "GETUTCDATE" -> FunDef(0, 0),
+    "ISDATE" -> FunDef(1, 1),
+    "MONTH" -> FunDef(1, 1),
+    "SMALLDATETIMEFROMPARTS" -> FunDef(5, 5),
+    "SWITCHOFFSET" -> FunDef(2, 2),
+    "SYSDATETIME" -> FunDef(0, 0),
+    "SYSDATETIMEOFFSET" -> FunDef(0, 0),
+    "SYSUTCDATETIME" -> FunDef(0, 0),
+    "TIMEFROMPARTS" -> FunDef(5, 5),
+    "TODATETIMEOFFSET" -> FunDef(2, 2),
+    "YEAR" -> FunDef(1, 1),
+    "MIN_ACTIVE_ROWVERSION" -> FunDef(0, 0),
+    "NULLIF" -> FunDef(2, 2),
+    "IFF" -> FunDef(3, 3),
+    "ISJSON" -> FunDef(1, 2),
+    "JSON_VALUE" -> FunDef(2, 2),
+    "JSON_QUERY" -> FunDef(2, 2),
+    "JSON_MODIFY" -> FunDef(3, 3),
+    "JSON_PATH_EXISTS" -> FunDef(2, 2),
+    "ABS" -> FunDef(1, 1),
+    "ACOS" -> FunDef(1, 1),
+    "ASIN" -> FunDef(1, 1),
+    "ATAN" -> FunDef(1, 1),
+    "ATN2" -> FunDef(2, 2),
+    "CEILING" -> FunDef(1, 1),
+    "COS" -> FunDef(1, 1),
+    "COT" -> FunDef(1, 1),
+    "DEGREES" -> FunDef(1, 1),
+    "EXP" -> FunDef(1, 1),
+    "FLOOR" -> FunDef(1, 1),
+    "LOG" -> FunDef(1, 2),
+    "LOG10" -> FunDef(1, 1),
+    "PI" -> FunDef(0, 0),
+    "POWER" -> FunDef(2, 2),
+    "RADIANS" -> FunDef(1, 1),
+    "RAND" -> FunDef(0, 1),
+    "ROUND" -> FunDef(2, 3),
+    "SIGN" -> FunDef(1, 1),
+    "SIN" -> FunDef(1, 1),
+    "SQRT" -> FunDef(1, 1),
+    "SQUARE" -> FunDef(1, 1),
+    "TAN" -> FunDef(1, 1),
+    "GREATEST" -> FunDef(1, Int.MaxValue),
+    "LEAST" -> FunDef(1, Int.MaxValue),
+    "CERTENCODED" -> FunDef(1, 1),
+    "CERTPRIVATEKEY" -> FunDef(2, 3),
+    "CURRENT_USER" -> FunDef(0, 0),
+    "DATABASE_PRINCIPAL_ID" -> FunDef(0, 1),
+    "HAS_DBACCESS" -> FunDef(1, 1),
+    "HAS_PERMS_BY_NAME" -> FunDef(4, 5),
+    "IS_MEMBER" -> FunDef(1, 1),
+    "IS_ROLEMEMBER" -> FunDef(1, 2),
+    "IS_SRVROLEMEMBER" -> FunDef(1, 2),
+    "LOGINPROPERTY" -> FunDef(2, 2),
+    "ORIGINAL_LOGIN" -> FunDef(0, 0),
+    "PERMISSIONS" -> FunDef(0, 2),
+    "PWDENCRYPT" -> FunDef(1, 1),
+    "PWDCOMPARE" -> FunDef(2, 3),
+    "SESSIONPROPERTY" -> FunDef(1, 1),
+    "SUSER_ID" -> FunDef(0, 1),
+    "SUSER_NAME" -> FunDef(0, 1),
+    "SUSER_SID" -> FunDef(0, 2),
+    "SUSER_SNAME" -> FunDef(0, 1),
+    "USER_ID" -> FunDef(0, 1),
+    "USER_NAME" -> FunDef(0, 1))
```

Comment:
should we have an IR for holding un-covered expression so we can transpile as is on target side.

Reply:
> should we have an IR for holding un-covered expression so we can transpile as is on target side.

I am going to close this PR and resubmit it without the fork. 

On functions that are not covered - we may also want to collect those as lint messages (see other PR about collecting messages), but also it needs to check for user defined functions, but that is functionality for another PR I think.